docs: fix instructions for debugging a karma test (#44244)
angular/angular: docs/BAZEL.md

### Debugging a Karma Test

- Run test: `yarn bazel run packages/core/test:test_web_chromium` or `yarn bazel run packages/core/test:test_web_firefox`

> - Open chrome or firefox at: [http://localhost:9876/debug.html](http://localhost:9876/debug.html)

Author: sorry, I just tried with Chromium and figured that `test_web_chromium` was for chrome and `test_web_firefox` for firefox and that those were the only two allowed (each for its respective command) :sweat_smile:

> - Open chrome or firefox at: [http://localhost:9876/debug.html](http://localhost:9876/debug.html)

Author: I went for `any browser` I hope it's fine :slightly_smiling_face:

> - Open chrome or firefox at: [http://localhost:9876/debug.html](http://localhost:9876/debug.html)

Reviewer: Perfect!

> - Open the browser's console to see the tests run

Reviewer: The wording wasn't great, but what this meant was that you can open DevTools to debug the tests (for example, you can go to the `Sources` panel and set break-points, etc.).

But argueably, the easiest way to go about debugging the tests is setting a `debugger` statement in the test code and open DevTools.

So, I suggest re-wording this a bit either making it more generic (something like "Open the browser's DevTools to debug the tests" - i.e. assuming the user knows how to debug code with DevTools) or mentioning setting the `debugger` statement.
(This will make it more obvious to users that are not familiar with debugging code/tests in the browser that there is more to it than just opening the Console and seeing the tests running.)

Author: Sure I'm making the change :slightly_smiling_face: 

.... is it ok to avoid saying DevTools? :sweat_smile:  I don't like it too much as lots of stuff fall under those not just the js console :sweat_smile: 

.... Open the browser's DevTools console? :thinking:

Reviewer: > .... is it ok to avoid saying DevTools? 😅 I don't like it too much as lots of stuff fall under those not just the js console 😅

Yeah, that's my whole point. To debug the tests you often use more than just the Console :grin:

Author: ah.... ok I can update the text with DevTools then :slightly_smiling_face:

Author: done :slightly_smiling_face:

> - Open the browser's DevTools to debug the tests (for example by focusing on specific tests via `fit` and/or `fdescribe` and also by adding `debugger` statements in them)

Reviewer: Great job adding the extra info and mentioning things such as focused tests/suites.
I think the way it is phrased it might be a little confusing for beginners, because it makes it sound like you could focus tests or add `debugger` statements in the DevTools.

How about something like:

```suggestion
- Open the browser's DevTools to debug the tests (after, for example, having focused on specific tests via `fit` and/or `fdescribe` or having added `debugger` statements in them)
```

> - Open the browser's DevTools console to debug the tests (for example by focusing on specific tests via `fit` and/or `fdescribe` and also by adding `debugger` statements in them)

Author: @gkalpak what do you think? :thinking: 

I am sort of thinking that I am stating the obvious.... but at least it is very beginner friendly, right? :thinking:
